diff -ur cricket-1.0.3/collect-subtrees cricket-1.0.3-local/collect-subtrees
--- cricket-1.0.3/collect-subtrees	Thu Oct 25 00:51:27 2001
+++ cricket-1.0.3-local/collect-subtrees	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/collector cricket-1.0.3-local/collector
--- cricket-1.0.3/collector	Tue Nov  6 01:04:47 2001
+++ cricket-1.0.3-local/collector	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/compile cricket-1.0.3-local/compile
--- cricket-1.0.3/compile	Fri Jun 29 01:47:02 2001
+++ cricket-1.0.3-local/compile	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/grapher.cgi cricket-1.0.3-local/grapher.cgi
--- cricket-1.0.3/grapher.cgi	Sat Nov  3 03:26:40 2001
+++ cricket-1.0.3-local/grapher.cgi	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/lib/Common/Map.pm cricket-1.0.3-local/lib/Common/Map.pm
--- cricket-1.0.3/lib/Common/Map.pm	Thu Oct 25 02:48:15 2001
+++ cricket-1.0.3-local/lib/Common/Map.pm	Thu Jun 20 17:52:01 2002
@@ -17,7 +17,15 @@
 #    You should have received a copy of the GNU General Public License
 #    along with this program; if not, write to the Free Software
 #    Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
-
+#
+#    BUGS : If you try to get an interface description from windows computers
+#           it returns a 0 byte at the end.
+#           ex : snmpwalk windowshost public 1.3.6.1.2.1.2.2.1.2
+#                returns : MS TCP Loopback interface00
+#           and when you try to compare this with your config file it fails.
+#	    The problem has been fixed  with :$name  =~ s/\000//g; in line 169. 
+#        
+#
 package Common::Map;
 
 use snmpUtils;
@@ -156,6 +164,9 @@
         my($row);
         foreach $row (@ret) {
             my($inst, $name) = split(':', $row, 2);
+	    #Windows interfaces returns a 0 byte at the enda.
+	    #The sentence above fix this problem.
+	    $name  =~ s/\000//g;
             $main::gMapCache{$snmp}->{$mapkey}->{$name} = $inst;
         }
     }
diff -ur cricket-1.0.3/mini-graph.cgi cricket-1.0.3-local/mini-graph.cgi
--- cricket-1.0.3/mini-graph.cgi	Thu Oct 25 00:42:55 2001
+++ cricket-1.0.3-local/mini-graph.cgi	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/subtree-sets cricket-1.0.3-local/subtree-sets
--- cricket-1.0.3/subtree-sets	Tue Feb  8 10:58:14 2000
+++ cricket-1.0.3-local/subtree-sets	Mon Jun 17 10:51:17 2002
@@ -10,5 +10,4 @@
 logdir:	cricket-logs
 
 set normal:
-	/routers
-	/router-interfaces
+	/servers
diff -ur cricket-1.0.3/util/dump-targets cricket-1.0.3-local/util/dump-targets
--- cricket-1.0.3/util/dump-targets	Sun Jul 22 01:57:33 2001
+++ cricket-1.0.3-local/util/dump-targets	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/util/generate-statics cricket-1.0.3-local/util/generate-statics
--- cricket-1.0.3/util/generate-statics	Mon Nov  5 23:54:01 2001
+++ cricket-1.0.3-local/util/generate-statics	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/util/get-collector-stats cricket-1.0.3-local/util/get-collector-stats
--- cricket-1.0.3/util/get-collector-stats	Wed Aug 22 00:28:03 2001
+++ cricket-1.0.3-local/util/get-collector-stats	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w 
+#!/usr/bin/perl -w 
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/util/listInterfaces cricket-1.0.3-local/util/listInterfaces
--- cricket-1.0.3/util/listInterfaces	Sun Jul 22 01:57:33 2001
+++ cricket-1.0.3-local/util/listInterfaces	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/util/newsstats cricket-1.0.3-local/util/newsstats
--- cricket-1.0.3/util/newsstats	Tue Feb  8 10:58:15 2000
+++ cricket-1.0.3-local/util/newsstats	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 ##
 # Get simple stats from INND with cleanfeed
diff -ur cricket-1.0.3/util/pmlines.pl cricket-1.0.3-local/util/pmlines.pl
--- cricket-1.0.3/util/pmlines.pl	Wed May 17 20:35:47 2000
+++ cricket-1.0.3-local/util/pmlines.pl	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl
+#!/usr/bin/perl
 
 # pmlines.pl
 #
diff -ur cricket-1.0.3/util/rrd-dump cricket-1.0.3-local/util/rrd-dump
--- cricket-1.0.3/util/rrd-dump	Sun Jul 22 01:57:33 2001
+++ cricket-1.0.3-local/util/rrd-dump	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w 
+#!/usr/bin/perl -w 
 # -*- perl -*-
 
 # this is a little script to exercise RRD::File -- useful
diff -ur cricket-1.0.3/util/rrd-tune cricket-1.0.3-local/util/rrd-tune
--- cricket-1.0.3/util/rrd-tune	Wed Aug 15 01:15:38 2001
+++ cricket-1.0.3-local/util/rrd-tune	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/util/run-subtree cricket-1.0.3-local/util/run-subtree
--- cricket-1.0.3/util/run-subtree	Tue May 23 02:18:35 2000
+++ cricket-1.0.3-local/util/run-subtree	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # Cricket: a configuration, polling and data display wrapper for RRD files
diff -ur cricket-1.0.3/util/test-url cricket-1.0.3-local/util/test-url
--- cricket-1.0.3/util/test-url	Tue Feb  8 10:58:15 2000
+++ cricket-1.0.3-local/util/test-url	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 #	 test-url: a simple script to see how fast a webserver is
diff -ur cricket-1.0.3/util/usrModemUsage cricket-1.0.3-local/util/usrModemUsage
--- cricket-1.0.3/util/usrModemUsage	Sun Jul 22 01:57:33 2001
+++ cricket-1.0.3-local/util/usrModemUsage	Mon Jun 17 08:58:16 2002
@@ -1,4 +1,4 @@
-#!/usr/local/bin/perl -w
+#!/usr/bin/perl -w
 # -*- perl -*-
 
 # usrModemUsage
